### PR TITLE
MNT: add gh-pages and * branch protection rules, extend AddBranchProtection options

### PR DESCRIPTION
**Pull request by tangkong**
## Description
- adds `restrictsPushes` and `blocksCreation` to the graphql createBranchProtectionRule
  - `restrictsPushes` was missing from the graphql side, despite existing in python (being actually applied now changes the base "master" branch protection rule
  - 
image

- adds branch protection rules for `gh-pages` branch and `*` (restricts the creation of new branches)
  - see [pcds-ci-test-repo](https://github.com/pcdshub/pcds-ci-test-repo-python/settings/branches) 's branch protection as an example
- adds the `--write` flag to update_github_settings
- adds a readme (ripped from confluence)

## Screenshots
What an approved PR looks like for a non-admin user
image

### CODE REVIEW

**Comment by tangkong** [comment]
This probably can wait until Ken gets back.  I've just kinda hacked around until I found something that works, maybe I misused something.

**Comment by ZLLentz** [comment]
I think this is all correct and clean, but I'm not intimately familiar with the graphql stuff

**Comment by tangkong** [comment]
We may want to consider alternate options for protections while we're here.  

We can create a team within pcdshub and add them as code-owners for all our repos (this will have to be applied across many repos)
We could dismiss reviews on new commits (I think this is a good option no matter what)

File: README.rst
Comment (klauer) [inline README.rst:1]: Awesome!

I totally did not recall writing some/most of this - good call adding it to the README

**Review by klauer — APPROVED** [review]
I think this is a good step forward as-is


>   We can create a team within pcdshub and add them as code-owners for all our repos (this will have to be applied across many repos)

I don't think this will be necessary until our development load is more balanced among the teams


> We could dismiss reviews on new commits (I think this is a good option no matter what)

I think we should do this one too

Additionally, we should probably have a list of repos we want these settings to apply to listed here (or somewhere else)
plc-summary has this: https://github.com/pcdshub/plc-summary/blob/master/add_submodules.sh
pcds-envs has this (kinda what we want): https://github.com/pcdshub/pcds-envs/blob/8481654b573f088c9a2bf8bf87fd1c9dd63028ce/envs/pcds/keep-updated.txt#L1-L35

**Pull request merged**
